integration/tests/basic/test_rpc_calls.py

```diff
@@ -174,9 +175,55 @@ def test_eth_gas_price(self, param: tp.Union[str, None]):
             response["result"]
         ), f"Invalid current gas price `{response['result']}` in wei"
 
+    @pytest.mark.parametrize(
+        "blockhash, tag1, tag2", [
+            (True, None, None),
+            (True, Tag.EARLIEST, None),
+            (True, None, Tag.LATEST),
+        ]
+    )
+    def test_eth_get_logs_blockhash(self, event_caller_contract, blockhash, tag1, tag2):
+        number = random.randint(1, 100)
+        text = "".join([random.choice(string.ascii_uppercase) for _ in range(5)])
+        bytes_array = text.encode().ljust(32, b'\0')
+        bol = True
+        tx = self.make_tx_object()
+        instruction_tx = event_caller_contract.functions.allTypes(
+            self.sender_account.address, number, text, bytes_array, bol
+        ).build_transaction(tx)
+        receipt = self.web3_client.send_transaction(self.sender_account, instruction_tx)
+
+        params = {}
+        receipt_blockhash = receipt["blockHash"].hex()
+        if blockhash:
+            params["blockHash"] = receipt_blockhash
+        if tag1:
+            params["fromBlock"] = tag1.value
+        if tag2:
+            params["toBlock"] = tag2.value
+
+        response = self.proxy_api.send_rpc("eth_getLogs", params=params)
+        if blockhash and not tag1 and not tag2:
+            assert "error" not in response
+            assert response["result"][0]["blockHash"] == receipt_blockhash
+            assert_fields_are_hex(response["result"][0],
+                                  ["transactionHash", "blockHash",
+                                   "blockNumber", "transactionIndex", "address",
+                                   "logIndex", "data", "transactionLogIndex"])
+            assert_fields_are_boolean(response["result"][0], ["removed"])
+        else:
+            pass
+            # NDEV-2237 bug
```

Comment:
maybe better to move it to separeted test? 
It is not good to have tests with if/else (dependences of responce)

Reply:
Done